Tests/AppTests/GithubTests.swift

```diff
@@ -118,143 +114,133 @@ class GithubTests: AppTestCase {
+                      isDraft: false,
+                      publishedAt: iso8601.date(from: "2021-04-22T02:50:05Z")!,
+                      tagName: "5.4.3",
+                      url: "https://github.com/Alamofire/Alamofire/releases/tag/5.4.3")
+            ))
+            XCTAssertEqual(res.repository?.repositoryTopics.totalCount, 15)
+            XCTAssertEqual(res.repository?.repositoryTopics.nodes.first?.topic.name,
+                           "networking")
+            XCTAssertEqual(res.repository?.stargazerCount, 35831)
+            XCTAssertEqual(res.repository?.isInOrganization, true)
+            XCTAssertEqual(res.repository?.homepageUrl, "https://swiftpackageindex.com/Alamofire/Alamofire")
+            // derived properties
+            XCTAssertEqual(res.repository?.lastIssueClosedAt,
+                           iso8601.date(from: "2021-06-09T00:59:39Z"))
+            // merged date is latest - expect that one to be reported back
+            XCTAssertEqual(res.repository?.lastPullRequestClosedAt,
+                           iso8601.date(from: "2021-06-07T22:47:01Z"))
+        }
     }
 
     func test_fetchMetadata_badRequest() async throws {
         Current.githubToken = { "secr3t" }
-        let client = MockClient { _, resp in
-            resp.status = .badRequest
-        }
-
-        do {
-            _ = try await Github.fetchMetadata(client: client,
-                                               owner: "alamofire",
-                                               repository: "alamofire")
-            XCTFail("expected error to be thrown")
-        } catch {
-            guard case Github.Error.requestFailed(.badRequest) = error else {
-                XCTFail("unexpected error: \(error.localizedDescription)")
-                return
-            }
-        }
-    }
 
-    func test_fetchMetadata_badUrl() async throws {
-        let pkg = Package(url: "https://foo/bar")
-        let client = MockClient { _, resp in
-            resp.status = .ok
-        }
-        do {
-            _ = try await Github.fetchMetadata(client: client, packageUrl: pkg.url)
-            XCTFail("expected error to be thrown")
-        } catch {
-            guard case Github.Error.invalidURL = error else {
-                XCTFail("unexpected error: \(error.localizedDescription)")
-                return
+        await withDependencies {
+            $0.httpClient.post = { @Sendable _, _, _ in .badRequest }
+        } operation: {
+            do {
+                _ = try await Github.fetchMetadata(owner: "alamofire",
+                                                   repository: "alamofire")
+                XCTFail("expected error to be thrown")
+            } catch {
+                guard case Github.Error.requestFailed(.badRequest) = error else {
+                    XCTFail("unexpected error: \(error.localizedDescription)")
+                    return
+                }
             }
         }
     }
```

Comment:
This test is now superfluous. It tested the `parseOwnerName` part of the `static func fetchMetadata(client: Client, packageUrl: String) async throws → Metadata` overload that was actually not used and which I removed.